File: iterator/c-sharp/EmployeeListAggregate.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Iterator
{
    // Aggregate (IEmployeeAggregate) arayüzünü uygular ve Iterator arayüzünü uygulayan sınıf örneği üretir.
    // Değerlerin referansları burada bulunur.
    // UML diyagramındaki ConcreteAggregate  sınıfına denk gelmektedir.
    class EmployeeListAggregate : IEmployeeAggregate
    {
        private List<Employee> _employees;

        public EmployeeListAggregate()
        {
            _employees = new List<Employee>();
        }

        public void Add(Employee employee)
        {
            _employees.Add(employee);
        }

        public int Count()
        {
            return _employees.Count;
        }

        public Employee Get(int index)
        {
            return _employees[index];
        }

        public IIterator<Employee> CreateIterator()
        {
            return new EmployeeIterator(this);
        }
    }
}

File: iterator/go/employee-list-aggregate.go
package main

// Aggregate (IEmployeeAggregate) arayüzünü uygular ve Iterator arayüzünü uygulayan sınıf örneği üretir.
// Değerlerin referansları burada bulunur.
// UML diyagramındaki ConcreteAggregate  sınıfına denk gelmektedir.
type employeeListAggregate struct {
	employees []*Employee
}

func NewEmployeeListAggregate() *employeeListAggregate {
	return &employeeListAggregate{
		employees: make([]*Employee, 0),
	}
}

func (e *employeeListAggregate) Add(employee *Employee) {
	e.employees = append(e.employees, employee)
}

func (e *employeeListAggregate) Count() int {
	return len(e.employees)
}

func (e *employeeListAggregate) CreateIterator() Iterator {
	return NewEmployeeIterator(e)
}
